src/tests/system/tests/test_netgroups.py

```diff
@@ -108,3 +124,282 @@ def test_netgroups__add_remove_netgroup_member(client: Client, provider: Generic
+    :setup:
+        1. Create users, groups and start sssd.
+    :steps:
+        1. Verify if the tuple (testhost5,ng9005,ldap.test) exists in the "DEVUsers" netgroup,
+            using the getent netgroup command.
+        2. Check if the combination of host (testhost5), user (ng9005), and domain (ldap.test)
+            is correctly part of the "DEVUsers" netgroup using libc.innetgr.
+    :expectedresults:
+        1. Tuple (testhost5,ng9005,ldap.test) exists in the "DEVUsers" netgroup
+        2. Combination of host (testhost5), user (ng9005), and domain (ldap.test)
+            is correctly part of the "DEVUsers" netgroup
+    :customerscenario: False
+    """
+    ou = ldap.ou("Netgroup")
+    create_users(ldap)
+
+    qa_users = ldap.netgroup("QAUsers", basedn=ou).add()
+    qa_users.add_member(host="testhost1", user="ng9000", domain="ldap.test")
+    dev_users = ldap.netgroup("DEVUsers", basedn=ou).add()
+    dev_users.add_member(host="testhost5", user="ng9005", domain="ldap.test")
+
+    client.sssd.dom("test")["entry_cache_timeout"] = "60"
+    client.sssd.start()
+
+    assert "(testhost5,ng9005,ldap.test)" in client.tools.getent.netgroup("DEVUsers").members
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_netgroup__membernisnetgroup(client: Client, ldap: LDAP):
+    """
+    :title: Add more complex LDAP netgroup structure by nesting one netgroup within another.
+    :setup:
+        1. Create users, groups and start sssd.
+    :steps:
+        1. Check that (testhost5, ng9005, ldap.test) is present as a direct member of "DEVUsers".
+        2. Check that (testhost1, ng9000, ldap.test) is also present,
+            even though this tuple was added to "QAUsers", not "DEVUsers".
+            This confirms that the nested group membership is working correctly
+            (since "QAUsers" is nested within "DEVUsers").
+    :expectedresults:
+        1. (testhost5, ng9005, ldap.test) is present as a direct member of "DEVUsers".
+        2. (testhost1, ng9000, ldap.test) is present as a direct member of "DEVUsers".
+    """
+    ou = ldap.ou("Netgroup")
+    create_users(ldap)
+
+    qa_users = ldap.netgroup("QAUsers", basedn=ou).add()
+    qa_users.add_member(host="testhost1", user="ng9000", domain="ldap.test")
+
+    dev_users = ldap.netgroup("DEVUsers", basedn=ou).add()
+    dev_users.add_member(host="testhost5", user="ng9005", domain="ldap.test")
+    ldap.ldap.modify(dev_users.dn, add={"memberNisNetgroup": "QAUsers"})
+
+    client.sssd.dom("test")["entry_cache_timeout"] = "60"
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(testhost5, ng9005, ldap.test)" in member
+    assert "(testhost1, ng9000, ldap.test)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_netgroup__add_dn_membernisnetgroup(client: Client, ldap: LDAP):
+    """
+    :title: Adding dn to memberNisNetgroup
```

Comment:
The test is to assert that SSSD processes 'memberNisNetgroup' attribute is the DN of a group. A couple of  things, 

1. The word, 'check' is like assert; you are 'adding' the netgroup and users, and you are not looking up the users. This setup is more complex than just one or two users. So it seems like the first step is to create the users and assert. 
2. Modifying the group object to use the DN needs to be a step.